internal/controller/controller_functions.go

```diff
@@ -74,51 +74,134 @@
 	return nil
 }
 
-// ProcessRestartRules processes restart rules for deployments
+// ProcessRestartRules processes restart rules for deployments with delay support
 func (r *BaseReconciler) ProcessRestartRules(ctx context.Context, restartRules []*karov1alpha1.RestartRule, resourceName, resourceType string) error {
 	logger := log.FromContext(ctx)
 
-	// For every restart rule returned by the store
+	// Group targets by workload key to handle multiple rules applying to the same target
+	type targetInfo struct {
+		target   karov1alpha1.TargetSpec
+		rules    []*karov1alpha1.RestartRule
+		maxDelay *time.Duration
+	}
+	targetMap := make(map[string]*targetInfo)
+
+	// First, collect all targets and find the maximum delay for each
 	for _, rule := range restartRules {
 		logger.Info("Processing restart rule",
 			"restartRule", rule.Name,
 			"namespace", rule.Namespace,
 			"resource", resourceName,
 			"resourceType", resourceType)
 
-		// Get all targets and iterate over them
 		for _, target := range rule.Spec.Targets {
-			// If the target is a Deployment, do a rollout restart
-			if target.Kind == "Deployment" {
-				if err := r.RestartDeployment(ctx, target, rule); err != nil {
-					logger.Error(err, "Failed to restart deployment",
-						"deployment", target.Name,
-						"resource", resourceName,
-						"resourceType", resourceType,
-						"restartRule", rule.Name)
-
-					// Record failed restart in status
-					if statusErr := r.recordRestartEvent(ctx, rule, target, resourceName, resourceType, "Failed", err.Error()); statusErr != nil {
-						logger.Error(statusErr, "Failed to record restart event")
-					}
-
-					continue
+			if target.Kind != "Deployment" {
+				continue // Only handle Deployment targets for now
+			}
+
+			targetNamespace := target.Namespace
+			if targetNamespace == "" {
+				targetNamespace = rule.Namespace
+			}
+			workloadKey := fmt.Sprintf("%s/%s", targetNamespace, target.Name)
```

Comment:
Why not use the target kind in the workload key ? In the issue it is explicitly stated, that it should work for workloads.